### DISCUSSION TITLE
why I failed to use a vector reference in a function?

### DISCUSSION THREAD

**Discussion by xiaoxiongli** [pybind/pybind11#4966] [General]
in C++ file:

void load_data2(std::vector<float>&  data)
{
    data[0] =2;
}
PYBIND11_MODULE(libmy_project, m)

in python file:
data = [0]
libmy_project.load_data2(data)
print(data)

the result is [0],   Why ?  I feel confuse why data not changed?

**Comment by wuguinianjing** [comment]
me too
